Assignment6/Array.cs

```
@@ -1,10 +1,220 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 
 namespace Assignment6
 {
-    public class Array<T>
+    public class Array<T> : ICollection<T>, IEnumerable<T> where T : IEquatable<T>
     {
+        public int Capacity { get; private set; }
+
+        private List<T> _Array;
         
+
+        public Array(int capacity)
+        {
+            if (capacity <= 0)
+            {
+                throw new IndexOutOfRangeException($"{nameof(capacity)} must be greater than 0");
+            }
+            Capacity = capacity;
+            _Array = new List<T>();
+        }
+
+        public Array()
+        {
+            _Array = new List<T>();
+        }
+
+        public int Count { get; private set; } = 0;
+
+        public bool IsReadOnly => throw new NotImplementedException();
+
+
+
+        public void Add(T item)
+        {
+            if (Count >= Capacity && Capacity != 0)
+            {
+                throw new IndexOutOfRangeException("Array is Full");
+            }
+            if (item == null)
+            {
+                throw new ArgumentNullException(nameof(item), "Cannot Add Null Item Reference");
+            }
+
+            _Array.Add(item);
+            Count++;
+
+
+
+        }
+
+        public void Clear()
+        {
+            _Array.Clear();
+            Count = 0;
+
+        }
+
+        public bool Contains(T item)
+        {
+            if (Count > Capacity)
+            {
+                throw new InvalidOperationException($"{nameof(item)} was not found");
+
+            }
+
+            if (_Array.Contains(item))
+            {
+                return true;
+            }
+
+            throw new InvalidOperationException($"{nameof(item)} was not found");

```

Why throw an exception rather than return false?  What is the point of the return if you always return true or throw an exception?